### PR TITLE
Adding specs to components

### PR DESCRIPTION
**Pull request by tetriscode**
## Status
**READY**

## JIRA Ticket
SPACON-415

## Description
Adding specs and unit test to rest endpoints

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] License

## Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

```sh
lein run -m spacon.generate-data
lein test
```

@boundlessgeo/spatial-connect

### CODE REVIEW

File: server/src/spacon/components/device/db.clj [hunk 1]
```diff
@@ -45,10 +47,11 @@
...
     (s/explain-str :spacon.specs.device/device-spec d)
-    (entity->map
-     (create-device<! {:identifier  (:identifier d)
-                       :name        (:name d)
-                       :device_info (json/write-str (:device-info d))}))))
+    (if-let [dev (create-device<! {:identifier  (:identifier d)
+                                   :name        (:name d)
+                                   :device_info (json/write-str (:device-info d))})]
+      (sanitize dev)
+      (sanitize d))))
```
Comment (mrcnc) [inline server/src/spacon/components/device/db.clj]: Seems like you want either `(sanitize dev)` or `(sanitize d)` but not both.

File: server/src/spacon/specs/store.clj [hunk 2]
```diff
@@ -13,13 +13,13 @@
 (s/def :store/id (s/with-gen
                    (s/and string? #(re-matches uuid-regex %))
                    #(uuid-string-gen)))
-(s/def :store/store_type string?)
+(s/def :store/store-type string?)
 (s/def :store/version string?)
 (s/def :store/uri string?)
 (s/def :store/name string?)
 (s/def :store/team-id (s/with-gen (s/and int? pos?)
```
Comment (tetriscode) [inline server/src/spacon/specs/store.clj]: pos-int

File: server/test/spacon/team_test.clj [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
+(deftest team-get-test []
+                   (let [res (utils/request-get "/api/teams")]
+                     (is (some? (:result res)))))
 
-(deftest all-team-test
-  (is (true? (spec-passed? `teams/all))))
+(deftest team-get-one-test []
+                           (let [res (utils/request-get "/api/teams")
+                                 team (utils/request-get (str "/api/teams/" (-> res :result first :id)))]
+                             (is (some? team))))
```
Comment (mrcnc) [inline server/test/spacon/team_test.clj]: Check that the response matches the spec

**Comment by mrcnc** [comment]
👍

**Pull request merged**